identity/aziot-identityd/src/identity.rs

```diff
@@ -684,10 +686,15 @@ impl IdentityManager {
         local_gateway_hostname: Option<String>,
         payload: Option<Payload>,
     ) -> Result<IoTHubDevice, Error> {
-        let backup_device = self.get_backup_provisioning_info(credentials.clone());
+        let Some(backup_device) = self.get_backup_provisioning_info(credentials.clone()) else {
+            return Err(Error::DpsClient(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "backup device cannot be none",
+            )));
+        };
```

Comment:
Where did this come from? The original code would do the DPS request.

Reply:
The linter complained the code could have been `let-else` instead of `let` then `backup_device.is_some()` with `backup_device.expect("backup device cannot be none")`

Reply:
Sure. That doesn't mean you can make changes to the actual behavior of the code.

Reply:
```rust
if skip_if_backup_is_valid {
    if let Some(backup_device) = backup_device {
        log::info!(...);

        return Ok(backup_device);
    }
}

let dps_request = ...
```

Reply:
Wilco